development-workflow.md

```diff
@@ -0,0 +1,60 @@
+
+## 🔮 Roadmap
+The MeiliSearch roadmap is public on [roadmap.meilisearch.com](https://roadmap.meilisearch.com).
+Anyone can vote for future features or submit an new idea. The roadmap is moderated and prioritized by MeiliSearch CEO.
+When an idea on the roadmap is prioritized it will enter in the development workflow.
+
+> __Note__
+> Not all specifications must be born out of a card on the roadmap and not all roadmap cards will be transformed into specifications.
+
+
+## 🎼 Development Workflow
+Many changes including bug fixes or documentation improvements can be implemented and reviewed via the normal GitHub pull request workflow which won't be covered in this document.
+
+Some changes are more substantial and we ask these to be put through a design process and produce consensus among the MeiliSearch community  and teams. 
+As a rule of thumb, any changes that impact in any way the user of MeiliSearch (without it being a bugfix) should follow this workflow.
+
+### 📝 Specifications (or RFCs) process
+
+We use specification instead of RFCs in our context because RFCs are more open to discussion (comments) than specifications. As MeiliSearch is still a young project and the founding team have a pretty clear idea about where MeiliSearch is heading we hope this process will help us to be better at specifying MeiliSearch rather than having long discussions about such design or such feature.
+
+That being said, this repository is public because anyone is more than welcome to participate!
+
+---
+A specification is a document that lives in the [meilisearch/specifications](meilisearch/specifications) repository:
+- A specification open to discussion is materialized as an open pull request.
+
+- An "active" specification is materialized as a merged specification the main branch.
+    - _Once a specification is "active", its authors or other contributors may implement and submit the feature as a pull request to the MeiliSearch repository. Being "active" does not mean that the feature will be utimately merged; it does mean that in principle all major stakeholders have agreed to the feature and are amenable to merging it._
+
+- An "implemented" specification define the current behavior of the search engine and must be marked as "implemented".
+
+### 🏗 Implementation process
+When a specification is "active", it can be implemented by any contributor to the repository.
+If a specification cannot be implemented in one pull request, a tracking issue linked to the specification must be written in order to trackm down the different pull request needed for that specification.
+
+Every pull request on the repository must be linked to a tracking issue or to a specification when possible.
+Regularly, pull request and issues on the main repository should be submitted for sorting in order to keep these manageable.
```

Comment:
In some cases, I have to create multiple issues, in one repo or even in several repos. I think we should re-write this paragraph then :)
I suggest (not definitive):
- the spec is validate
- the core team manager open the issue(s)
- the core team (or the contrib) implements the issues
- When all the issues are closed (because the PR are merged), the spec is considered as implemented

It's easier for me to open the issue(s) according to the spec (and link the spec in the issues) to track the job. Also, it's easier for us (the core team) to link an issue rather than a spec in a PR.
But if you think this is really important to track the PR in the spec, let me know, we can find another way to manage this!

Plus, do you need a label on the issues/PRs? Like `Tracking issue` or `Spec related`?

Reply:
Agreed! The implementation process should add that if it's easier for the delivery aspect.

It could be nice to add the merged PRs on the active spec when it goes to the `implemented` state. The comment section is still open on a closed pull request to add that but I don't think this is the best way to share information. 
Can we keep it simple at the moment since the related issues will at least be linked to the spec?

Also, it's important to keep in mind that an active spec (valid and merged) is `Ready To Be Implemented` before being Ìmplemented`and we can advocate to froze it at any moment for diverse reasons.

I vote for the `Spec Related` label if we need one, I find it much more precise.